### ISSUE TITLE
Discussion: How could we merge `EXASOL` and `EXASOL_FS`

### ISSUE THREAD

**Issue by sti0** [sqlfluff/sqlfluff#1495]
Hi there,

I like to discuss how we could merge the both dialects `EXASOL` and `EXASOL_FS`.

The only reason why Exasol was splitted in two dialects is, that scripts and functions are delimited by `/`-sign instead of `;`.
To solve this we could extend `DelimiterSegment` like this:

```
DelimiterSegment=OneOf(
        NamedParser(
            "function_script_terminator", CodeSegment, type="statement_terminator"
        ),
        Ref("SemicolonSegment"),
```

This would work for scripts but not for functions because the script body is parsed with `Nothing()` while function body is parsed by some `Segments`. Lines in the function body are delimited by `;` which was lexed as `statement_terminator`

```
CREATE OR REPLACE FUNCTION percentage ( fraction DECIMAL,
                                        entirety DECIMAL)
 RETURN VARCHAR(10)
 IS
    res DECIMAL;
BEGIN
    res := (100*fraction)/entirety ;
     RETURN res || ' %' ;
END percentage;
```
I tried to solve this by extending `FileSegment` with a `OneOf()` but this only works for the first type. Function or statement or vice versa (which stands in the first line)

```
parse_grammar = OneOf(
        Delimited(
            Ref("FunctionStatementSegment"),
            delimiter=Ref("FunctionScriptTerminatorSegment"),
            allow_gaps=True,
            allow_trailing=True,
        ),
        Delimited(
            Ref("StatementSegment"),
            delimiter=Ref("SemicolonSegment"),
            allow_gaps=True,
            allow_trailing=True,
        ),
    )
```

Does anyone has a solution for that?

**Comment by tunetheweb** [comment]
Would love for this to be solved!

What about something like this (untested):

```
parse_grammar = AnyNumberOf(
        Sequence(
            Ref("FunctionStatementSegment"),
            Ref("FunctionScriptTerminatorSegment"),
        ),
        Sequence(
            Ref("StatementSegment"),
            Ref("SemicolonSegment"),
        ),
    )
```

**Comment by sti0** [comment]
I tried this before but this ends in a `RuntimeError: Grammar refers to 'NameKeywordSegment' which was not found in the exasol dialect` exception when parsing `functions`. I can't unterstand why this happens. E.g. `SELECT` statements are parsed correctly.
I pushed my changes - a little bit messy atm - here: https://github.com/sti0/sqlfluff/tree/feature/merge-exasol-fs

**Comment by tunetheweb** [comment]
When you see that it, it means it is potentially using `"NAME"` somewhere and your dialect doesn't support that. Looks like it's in `TransactionStatementSegment` so setting that to `Nothing()` in Exasol, or just removing the `NAME` part from that should resolve that error (but potentially lead to another error for the next keyword you don't support - what fun!).

**Comment by sti0** [comment]
Thanks for the hint. I fixed the `TransactionStatementSegment` and the error goes away. The statement you mentioned pointed me in the right direction but there are two issues open (mentioned in #1498).

**Comment by sti0** [comment]
solved #1498
